DOC Add example for metrics.RocCurveDisplay

steinfurt opened the pull request:
#### Reference Issues/PRs
See also #3846

#### What does this implement/fix? Explain your changes.
Implements a simple example/doctest for metrics.RocCurveDisplay

#### Any other comments?

steinfurt committed:
Add examples for metrics.RocCurveDisplay and metrics.plot_roc_curve

thomasjpfan commented:
Thank you for the PR @steinfurt 

Since we recommend users to use `plot_roc_curve` to create this object, I would prefer an example in `plot_roc_curve` instead.

steinfurt commented:
Sure, I can add that. Should I remove the class doctest then? But since it is not "internal", I tend towards leaving it additionally...

thomasjpfan commented:
I am okay with including it. For curiosity, do you think you will find yourself creating this `Display` object directly? (As shown in this PR's example)

steinfurt commented:
Not sure, to be honest. But it may be somewhat convenient if the `fpr`, `tpr`, `thresholds` and `roc_auc` are needed afterwards (for logging etc.) and the display is only used temporarily during development. Of course, as implemented, the values could be obtained via `display.roc_auc` etc. after running `display = plot_roc_curve(...)`.

steinfurt commented:
Okay, CI includes environments which do not include `matplotlib`. That makes (doc-)testing a feature which imports it internally quite difficult. Should the `plot_roc_curve` example then be skipped altogether in doctests? Or how should one handle it, @thomasjpfan?

thomasjpfan commented on sklearn/metrics/_plot/roc_curve.py:
```diff
@@ -138,6 +153,21 @@ def plot_roc_curve(estimator, X, y, pos_label=None, sample_weight=None,
+    >>> clf = svm.SVC(random_state=0)
+    >>> clf.fit(X_train, y_train)
+    SVC(random_state=0)
+    >>> metrics.plot_roc_curve(estimator=clf, X=X_test, y=y_test)
```
I am happy with skipping this and removing the `pytest.importorskip`. Since `estimator`, `X`, `y` are position arguments, I would be okay with using positions here.

thomasjpfan commented on sklearn/metrics/_plot/roc_curve.py:
```diff
@@ -36,6 +36,21 @@ class RocCurveDisplay:
+    >>> fpr, tpr, thresholds = metrics.roc_curve(y, pred)
+    >>> roc_auc = metrics.auc(fpr, tpr)
+    >>> metrics.RocCurveDisplay(fpr=fpr, tpr=tpr, roc_auc=roc_auc,\
+                                estimator_name='example estimator').plot()
```
We can create the `Display` object and then call plot on another line. The plot to call will be skipped.

steinfurt committed:
Remove pytest.importorskip from examples

All calls related to matplotlib have to be skipped here, because in
general it need not be installed. This fact is also reflected by some
CI environments.

Although using 'doctest: +SKIP' instead of pytest.importorskip also
skips the call to metrics.plot_roc_curve during tests in environments
where matplotlib is in fact available, the main goal here is to have a
clean example.

thomasjpfan reviewed (approved):
Thank you @steinfurt

rth reviewed (approved):
Thanks @steinfurt !
